lib/ecto/adapters/sql.ex

```diff
@@ -514,22 +486,19 @@ defmodule Ecto.Adapters.SQL do
   @doc false
   def transaction(repo, opts, fun) do
     {pool_mod, pool, timeout} = repo.__pool__
-    opts            = Keyword.put_new(opts, :timeout, timeout)
-    timeout         = Keyword.fetch!(opts, :timeout)
+    opts    = Keyword.put_new(opts, :timeout, timeout)
+    timeout = Keyword.fetch!(opts, :timeout)
 
-    trans_fun = fn(ref, mode, depth, queue_time) ->
-      transaction(repo, ref, mode, depth, queue_time, timeout, opts, fun)
+    trans_fun = fn(ref, {mod, _conn}, mode, depth, queue_time) ->
+      transaction(repo, ref, mod, mode, depth, queue_time, timeout, opts, fun)
     end
 
     case Pool.transaction(pool_mod, pool, timeout, trans_fun) do
-      {:ok, {{:return, result}, entry}} ->
-        log(repo, entry)
```

Comment:
I am aware the goal here was to avoid logging inside the transaction but we already log the begin statement and all other queries inside the transaction, so logging the commit or rollback one is not a big deal and it cleans up the code considerably.